doc/source/theory/fluid_dynamics/linear_solvers.rst

```diff
@@ -1,4 +1,154 @@
+
+.. math::
+
+    r_k = b - \mathcal{A} x_k \in r_0 + \mathcal{A} \mathcal{K}_k (\mathcal{A}, r_0), \hspace{2cm} r_k \perp \mathcal{C}_k
+
+By knowing the properties of :math:`\mathcal{A}`, it is possible to determine constraints spaces :math:`\mathcal{C}_k` that lead to uniquely defined iterates :math:`x_k` for :math:`k=1,2,\dotsc`. The choice of :math:`\mathcal{C}_k` and subsequent implementation leads to a different method.
+
+
+Generalized Minimal Residual Method (GMRES)
+_____________________________________________
+
+If :math:`\mathcal{A}` is non-symmetric positive definite, an approximate solution can be found by choosing :math:`\mathcal{C}_k = \mathcal{K}_k(\mathcal{A}, r_0)`, one implementation of this corresponds to the GMRES method. 
+
+
+*  The GMRES method is based on what is called an Arnoldi iteration. This iteration has a :math:`k`-term recurrence, which means that its computational cost increases as :math:`k` increases. The method is not very practical for very large :math:`k` because of the increase on memory and computational requirements. Therefore, for these kinds of systems, usually a restart is needed after certain iterations (for example 30 iterations) to build a new Krylov subspace. This restart may affect the convergence of the method when the matrix is not positive definite; a remedy for this is the use of a preconditioner.
+
+* In Lethe, we use the Trilinos implementation of the GMRES method through `deal.II <https://www.dealii.org/>`_: `TrilinosWrappers::SolverGMRES <https://dealii.org/developer/doxygen/deal.II/classTrilinosWrappers_1_1SolverGMRES.html>`_. It is possible to specify the minimum residual, the solver residual, the maximum number of iterations and the maximum Krylov vectors. The latter is in fact the one that controls the number of iterations after which a restart should be done.
+
+Biconjugate Gradient Stabilized Method (BiCGStab)
+__________________________________________________
+
+One may choose instead  :math:`\mathcal{C}_k = \mathcal{K}_k(\mathcal{A}^T, r_0)`, which leads to the implementation of the BiCG method. However, this is not well defined for a general non-symmetric matrix :math:`\mathcal{A}`, therefore, a similar implementation with a stabilization step leads to the BiCGStab method. It avoids irregular convergence patterns by maintaining the convergence speed.
+
+* In the case of the BiCGStab method, the iteration is known as a transpose-free bi-Lanczos iteration. This kind of iteration has a fixed cost per iteration and no restart is ever needed. This method has a three-term recurrence, however, it needs two matrix-vector multiplications per iteration as it uses the system matrix twice to avoid the use of its transpose.  
```

Comment:
I would mention that using this linear solver with high-order elements does not work well.

Reply:
That's mostly a Sharp-Edge thing, it works well for the cases I tried with the regular solver ;)